pyclam/manifold.py

```diff
@@ -440,12 +444,40 @@ def _build_edges_matrix(self) -> None:
 
                 batch_left, batch_right = tuple(map(list, np.where(batch_differences <= 0.)))
                 [clusters[l_ + i].neighbors.update({clusters[r_]: batch_distances[l_, r_]}) for l_, r_ in zip(batch_left, batch_right) if (l_ + i) != r_]
+        return
+
+    def _build_edges_pruned(self) -> None:
+        """ Calculates edges for self by relying on tree-search. """
+        clusters: List[Cluster] = list(self.clusters.keys())
+        root = self.manifold.select('')
+
+        for cluster in clusters:
+            ancestor, radius = cluster, cluster.radius
+            while radius <= 0.:
+                ancestor, radius = ancestor.parent, ancestor.parent.radius
+
+            potential_neighbors: Dict['Cluster', Radius] = root.tree_search(
+                point=cluster.medoid,
+                radius=radius * 2 * np.sqrt(2),
```

Comment:
Weird not seeing dimensionality used here

Reply:
complexities cancel away